src/styles/index.scss

```diff
@@ -2,2 +1,57 @@
+@import 'vars';
+@import 'mixins';
+
+.page__body {
   margin: 0;
+  height: 100vh;
+
+  @include flex-block;
+}
+
+.calendar {
+  display: flex;
+  flex-flow: row wrap;
+  gap: $day-gap;
+  width: $calendar-width;
+  padding: 10px;
+
+  &__day {
+    @include flex-block;
+    @include text-style;
+
+    width: $day-width;
+    height: $day-width;
+    background-color: $calendar-color;
+    color: black;
+    border: $day-border;
+    transition-duration: 500ms;
+
+    &:hover {
+      cursor: pointer;
+      background-color: $pointer-calendar-color;
+      transform: translateY(-20px);
+    }
+
+    @for $numbers from 1 through 31 {
+      &:nth-child(#{$numbers})::before {
+        content: '#{$numbers}';
+      }
+    }
+  }
+
+  @each $day, $number in $days {
+    &--start-day-#{$day} {
+      .calendar__day:first-of-type {
+        margin-left: $number * ($day-width + $day-gap);
+      }
+    }
+  }
+
+  @for $numbers from 28 through 31 {
```

Comment:
The same applies here. Instead of hardcoding the number of days in a month, you can use a variable. This will make your calendar more flexible and easy to maintain.